dir.targets

```diff
@@ -44,9 +44,11 @@
           AfterTargets="_RestoreBuildToolsWrapper">
     <MSBuild Projects="$(RepoRoot)\targets\DeployDependencies.proj"
              Targets="DeployDependencies"
-             BuildInParallel="true"
-             Condition="'$(OsEnvironment)'=='Windows_NT'"
+             Properties="DeploymentPath=$(DeploymentDir)"
```

Comment:
Setting Properties in an `<MSBuild>` task is a scary thing.  In a multithreaded build, it creates a new set of Global Properties that can cause new instances of every project in the "cone of dependencies" to be built.  This often creates race conditions between distinct instances of the same project that write to the same file.

To use this, you need to guarantee that *every file written* in *any project transitively built by the referenced project* has a distinct path that depends on the property that changes here.  Oh, and you need to do that for all future file I/O, too--which makes this error prone :cold_sweat:.

This *can* be done safely (and in this case it looks OK), but because of the risks involved I usually caution against it.

The usual approach to avoiding these problems is to define the "function" that needs input from different projects in a `.targets` file and import that into the projects that will vary.  I'd recommend that in this case.

When there was a single deployment directory, it made sense to have a project reference to `DeployDependencies.proj`--that way the MSBuild engine would deduplicate requests to build it from many different projects and deploy the binaries only once.  Now, since you're creating distinct output directories for each project, I think it's a good idea to move that part into a `.targets`.

(THAT got longwinded, sorry!)